internal/controller/operatorconfigmap_controller.go

```diff
@@ -950,3 +1002,71 @@ func (c *OperatorConfigMapReconciler) deleteDelegatedCSI() error {
 	}
 	return nil
 }
+
+// TODO: Remove it in 4.20 release
+func (c *OperatorConfigMapReconciler) getCsiCustomTolerations() ([]corev1.Toleration, []corev1.Toleration, error) {
+	rookCephOperatorConfigName, csiPluginTolerationKey, csiProvisionerTolerationKey := "rook-ceph-operator-config", "CSI_PLUGIN_TOLERATIONS", "CSI_PROVISIONER_TOLERATIONS"
+	nodePluginTolerations, controllerPluginTolerations := []corev1.Toleration{}, []corev1.Toleration{}
+
+	// list the rook ceph cm
+	rookCephOperatorConfig := &corev1.ConfigMap{}
+	rookCephOperatorConfig.Name = rookCephOperatorConfigName
+	rookCephOperatorConfig.Namespace = c.OperatorNamespace
+
+	if err := c.get(rookCephOperatorConfig); err != nil {
+		if kerrors.IsNotFound(err) {
+			return nodePluginTolerations, controllerPluginTolerations, nil
+		}
+		return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to get Rook Ceph Operator ConfigMap: %v", err)
+	}
+
+	// Get the CSI tolerations from the Rook Ceph Operator ConfigMap
+	csiPluginTolerations, plugin := rookCephOperatorConfig.Data[csiPluginTolerationKey]
+	csiProvisionerToleration, provisioner := rookCephOperatorConfig.Data[csiProvisionerTolerationKey]
```

Comment:
why is it named `plugin` and `provisioner` aren't they booleans? or better only check for presence of value for the key?

Reply:
Ack